hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/ResultSetReturnImpl.java

```diff
@@ -53,6 +53,10 @@ public ResultSetReturnImpl(JdbcCoordinator jdbcCoordinator) {
 	@Override
 	public ResultSet extract(PreparedStatement statement) {
 		// IMPL NOTE : SQL logged by caller
+		long executeStart = 0;
+		if ( this.sqlStatementLogger.getLogSlowQuery() > 0 ) {
+			executeStart = System.currentTimeMillis();
```

Comment:
since you're measuring _elapsed time_ I would suggest to use nanoTime() instead.

Reply:
(which implies the configuration reading code might need to be adapted to deal with nano-seconds precision ?)

Reply:
I think milliseconds is enough. Also, thread-safe and more efficient.